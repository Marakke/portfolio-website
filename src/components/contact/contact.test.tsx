import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Contact from "./index";

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  contact: {
    titleWhite: "Test Title White",
    titleBlack: "Test Title Black",
    descriptionHeader: "Test Description Header",
    descriptionFooter: "Test Description Footer",
    highlightedWords: "HighlightedWordOne, HighlightedWordTwo",
    email: "email@email.com",
    linkedin: "linkedinaccount",
    github: "githubaccount",
  }
};

describe("Contact", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Contact component correctly", () => {
    const { container, getByTestId, getByRole } = render(<Contact />);
    expect(container).toMatchSnapshot();

    const contactContainer = getByTestId("contact-container");
    expect(contactContainer).toBeInTheDocument();

    const titleWhite = getByRole("heading", { name: "Test Title White Test Title Black" });
    expect(titleWhite).toBeInTheDocument();

    const anchorElements = container.querySelectorAll("a");
    expect(anchorElements.length).toEqual(3);

    expect(anchorElements[0]).toHaveTextContent("email@email.com");
    expect(anchorElements[1]).toHaveTextContent("linkedinaccount");
    expect(anchorElements[2]).toHaveTextContent("githubaccount");

    expect(anchorElements[0].href).toBe("mailto:email@email.com");
    expect(anchorElements[1].href).toBe("https://linkedinaccount/");
    expect(anchorElements[2].href).toBe("https://githubaccount/");

    const paragraphElements = container.querySelectorAll("p");
    expect(paragraphElements.length).toEqual(7);

    expect(paragraphElements[0].textContent).toBe("Test Description Header");
    expect(paragraphElements[1].textContent).toBe("Test Description Footer ");
  });
});
