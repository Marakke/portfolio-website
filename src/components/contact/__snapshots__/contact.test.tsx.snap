// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders the Contact component correctly 1`] = `
<div>
  <div
    class="contact-container"
    data-testid="contact-container"
    id="contact"
  >
    <div
      class="contact-title"
    >
      <h2
        class="title-white"
      >
        Test Title White
         
        <span
          class="title-black"
        >
          Test Title Black
        </span>
      </h2>
    </div>
    <div
      class="contact-content"
    >
      <div
        class="contact-links"
      >
        <div
          class="contact-link"
        >
          <img
            alt="Email Icon"
            src="test-file-stub"
          />
          <a
            class="link"
            href="mailto:email@email.com"
          >
            email@email.com
          </a>
        </div>
        <div
          class="contact-link"
        >
          <img
            alt="LinkedIn Icon"
            src="test-file-stub"
          />
          <a
            class="link"
            href="https://linkedinaccount"
            target="_blank"
          >
            linkedinaccount
          </a>
        </div>
        <div
          class="contact-link"
        >
          <img
            alt="GitHub Icon"
            src="test-file-stub"
          />
          <a
            class="link"
            href="https://githubaccount"
            target="_blank"
          >
            githubaccount
          </a>
        </div>
      </div>
      <div
        class="contact-description"
      >
        <p
          class="description"
        >
          Test Description Header
        </p>
        <p
          class="description"
        >
          Test 
          Description 
          Footer 
        </p>
      </div>
    </div>
    <div
      class="lines contact"
    >
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
      <p
        class="line"
      />
    </div>
  </div>
</div>
`;
